### PR TITLE
Adding bigquery logging

### PR DESCRIPTION
**Pull request by bridgetlane**

### CODE REVIEW

**Comment by bridgetlane** [comment]
Sorry, messed up my original PR by updating my branch to merge into our internal version of the provider. Heres the fixed up one

**Comment by bridgetlane** [comment]
Going to add the format field and a couple other fields to the UPDATE command, so hold off on merging for now https://github.com/sethvargo/go-fastly/pull/73

**Comment by bridgetlane** [comment]
This is ready to go if I can get a review on it

File: fastly/resource_fastly_service_v1_bigquerylogging_test.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  backend {
+    address = "%s"
+    name    = "tf -test backend"
+	}
+	
+	bigquerylogging {
+		name = "%s"
+		email = "email@example.com",
+		secret_key = "secretKey",
+    project_id = "example-gcp-project"
```
Comment (paultyng) [inline fastly/resource_fastly_service_v1_bigquerylogging_test.go]: Some mixed tabs/spaces here

File: fastly/resource_fastly_service_v1_bigquerylogging_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  }
+
+  backend {
+    address = "%s"
+    name    = "tf -test backend"
+  }
+
+	bigquerylogging {
+		name = "%s"
+    project_id = "example-gcp-project"
```
Comment (paultyng) [inline fastly/resource_fastly_service_v1_bigquerylogging_test.go]: Some mixed tabs/spaces here

**Review by paultyng — COMMENTED** [review]
Thanks for this! It looks good to me, just some minor whitespace issues in the acceptance tests, but I can clean that up if you don't have the time.

For running the acceptance testing, will this require some GCP infrastructure setup? Is the `secret_key` validated with Google synchronously or does it just accept it?

File: fastly/resource_fastly_service_v1.go [hunk 3]
```diff
@@ -2907,6 +3031,33 @@ func flattenGCS(gcsList []*gofastly.GCS) []map[string]interface{} {
 	return GCSList
 }
 
+func flattenBigQuery(bqList []*gofastly.BigQuery) []map[string]interface{} {
+	var BQList []map[string]interface{}
```
Comment (paultyng) [inline fastly/resource_fastly_service_v1.go:3043]: Minor style note, I wouldn't capitalize these as it makes them look exported.

**Comment by paultyng** [comment]
@bridgetlane I had to set a default on `format` to get the tests to pass, otherwise they would fail with a pending diff.

It seems like [per the docs](https://docs.fastly.com/api/logging#logging_bigquery) though that format may be required as the JSON schema needs to match your table?

> Apache style log formatting. Must produce JSON that matches the schema of your BigQuery table

I'm not really familiar with this, so would appreciate your input on how to handle.

**Comment by bridgetlane** [comment]
@paultyng sorry only just getting back to you on this. Thats odd, we don't provide a format when we use this functionality on our end (I rolled it into a custom provider and use it like so) and a default format is provided:

```
bigquerylogging {
    name = "logging_endpoint"
    project_id = "project"
    dataset = "some_dataset"
    table = "some_table"
  }

```
Depending on how you log, the format will or won't matter. I include a no_logging response condition on my logging (something I need to add to this provider) so we log manually rather than using the default logging that uses that formatting block. If you manually create a BigQuery logging endpoint in Fastly, the following is set by default:

```
{
  "timestamp":"%{begin:%Y-%m-%dT%H:%M:%S%z}t",
  "time_elapsed":%{time.elapsed.usec}V,
  "is_tls":%{if(req.is_ssl, "true", "false")}V,
  "client_ip":"%{req.http.Fastly-Client-IP}V",
  "geo_city":"%{client.geo.city}V",
  "geo_country_code":"%{client.geo.country_code}V",
  "request":"%{req.request}V",
  "host":"%{req.http.Fastly-Orig-Host}V",
  "url":"%{cstr_escape(req.url)}V",
  "request_referer":"%{cstr_escape(req.http.Referer)}V",
  "request_user_agent":"%{cstr_escape(req.http.User-Agent)}V",
  "request_accept_language":"%{cstr_escape(req.http.Accept-Language)}V",
  "request_accept_charset":"%{cstr_escape(req.http.Accept-Charset)}V",
  "cache_status":"%{regsub(fastly_info.state, "^(HIT-(SYNTH)|(HITPASS|HIT|MISS|PASS|ERROR|PIPE)).*", "\\2\\3") }V"
}
```

That should be set as the auto default, because the apache style formatting can't match the restricted BigQuery schema that it has to match (as far as I'm aware).

**Comment by bridgetlane** [comment]
@paultyng Also I assume you figured this out but it just blindly accepts the secret_key

**Comment by bridgetlane** [comment]
@paultyng I've put in a pr to add request logging to go-fastly create and update, should be an easy add to this PR once merged https://github.com/sethvargo/go-fastly/pull/75

**Comment by jonoden** [comment]
This would be helpful to merge, just pinging because I care. :)

**Comment by paultyng** [comment]
Acc test results: 

```shell
=== RUN   TestAccFastlyServiceV1_bigquerylogging_env
--- PASS: TestAccFastlyServiceV1_bigquerylogging_env (15.80s)
PASS
=== RUN   TestAccFastlyServiceV1_bigquerylogging
--- PASS: TestAccFastlyServiceV1_bigquerylogging (16.50s)
PASS
```

**Pull request merged**